verity/src/helm/templates/deployment.yaml

```diff
@@ -13,31 +13,51 @@ spec:
       maxUnavailable: 0
   selector:
     matchLabels:
-      app: {{ template "verity.fullname" . }}
+      {{- include "verity.selectorLabels" . | nindent 6 }}
   replicas: {{ .Values.startReplicas }}
   template:
     metadata:
       labels:
-        app: {{ template "verity.fullname" . }}
+        {{- include "verity.labels" . | nindent 8 }}
         revision: {{ .Release.Revision | quote }}
         actorSystemName: verity
-        tags.datadoghq.com/service: {{ .Values.service }}
-        tags.datadoghq.com/env: {{ .Values.env }}
+        {{- include "verity.datadogLabels" . | nindent 8 }}
+      {{- if .Values.deployment.vault.enable }}
+      annotations:
+        {{ include "verity.vaultAnnotations" . | indent 8 }}
+      {{- end }}
     spec:
+      # TODO parametrize afifnity better
+      # TODO why affinity might be needed ???
+      {{- if .Values.deployment.useAffinity }}
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: app.kubernetes.io/name
+                operator: In
+                values:
+                - {{ template "verity.name" . }}
+            # - key: app.kubernetes.io/instance
+            #   operator: In
+            #   values:
+            #   - {{ template "verity.instance-name" . }}
+      {{- end }}
```

Comment:
original Verity codebase doesn't include this section. If you are not sure if it is really necessary I would recommend to remove this section completely.

Reply:
removed